middlewares/osio/osio.go

```diff
@@ -99,85 +125,123 @@ func NewPreConfiguredOSIOAuth() *OSIOAuth {
 	if len(srvAccSecret) <= 0 {
 		panic("Missing SERVICE_ACCOUNT_SECRET")
 	}
-	return NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret)
+	return NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret, authTokenKey)
 }
 
-func NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret string) *OSIOAuth {
+func NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret, authTokenKey string) *OSIOAuth {
 	return &OSIOAuth{
 		RequestTenantLocation: CreateTenantLocator(http.DefaultClient, tenantURL),
-		RequestTenantToken:    CreateTenantTokenLocator(http.DefaultClient, authURL),
+		RequestTenantToken:    CreateTenantTokenLocator(http.DefaultClient, authURL, authTokenKey),
 		RequestSrvAccToken:    CreateSrvAccTokenLocator(authURL, srvAccID, srvAccSecret),
 		RequestSecretLocation: CreateSecretLocator(http.DefaultClient),
 		RequestTokenType:      CreateTokenTypeLocator(http.DefaultClient, authURL),
-		cache:                 &Cache{},
+		nsCache:               &Cache{},
+		tokenCache:            &Cache{},
 	}
 }
 
-func cacheResolverByID(tenantLocator TenantLocator, tokenLocator TenantTokenLocator, srvAccTokenLocator SrvAccTokenLocator, secretLocator SecretLocator, token string, tokenType TokenType, userID string) Resolver {
+func (a *OSIOAuth) resolveNamespaceByID(token string, userID string, params paramMap) Resolver {
```

Comment:
now resolveXXX() func made methods for OSIOAuth so no need to pass lot of parameter.